src/main/kotlin/org/jetbrains/research/ml/codetracker/reporting/GitHubErrorReporter.kt

```diff
@@ -64,32 +69,35 @@ class GitHubErrorReporter : ErrorReportSubmitter() {
 
     }
 
-    companion object {
-        private fun doSubmit(
-            event: IdeaLoggingEvent,
-            parentComponent: Component,
-            callback: Consumer<SubmittedReportInfo>,
-            errorInformation: ErrorInformation
-        ): Boolean {
-            if (event is IdeaReportingEvent) {
-                event.plugin?.let {
-                    errorInformation.setUserInformation(UserInformationType.PLUGIN_NAME, it.name)
-                    errorInformation.setUserInformation(UserInformationType.PLUGIN_VERSION, it.version)
-                }
+    private fun doSubmit(
+        event: IdeaLoggingEvent,
+        parentComponent: Component,
+        callback: Consumer<SubmittedReportInfo>,
+        errorInformation: ErrorInformation
+    ): Boolean {
+
+        if (event is IdeaReportingEvent) {
+            event.plugin?.let {
+                errorInformation.setUserInformation(UserInformationType.PLUGIN_NAME, it.name)
+                errorInformation.setUserInformation(UserInformationType.PLUGIN_VERSION, it.version)
             }
-            val dataContext = DataManager.getInstance().getDataContext(parentComponent)
-            val project = CommonDataKeys.PROJECT.getData(dataContext)
-            val task = ErrorReportTask(
-                project,
-                "Submitting error report",
-                true,
-                errorInformation,
-                CallbackWithNotification(callback, project)
-            )
-            project?.let { ProgressManager.getInstance().run(task) } ?: run {
+        }
+        val dataContext = DataManager.getInstance().getDataContext(parentComponent)
+        val project = CommonDataKeys.PROJECT.getData(dataContext)
+        val task = ErrorReportTask(
+            project,
+            "Submitting error report",
+            true,
+            errorInformation,
+            CallbackWithNotification(callback, project)
+        )
+        project.let {
+            if (it != null) {
+                ProgressManager.getInstance().run(task)
+            } else {
                 task.run(EmptyProgressIndicator())
             }
```

Comment:
just add '?' in `project.let` so you don't have to check if it is null inside the 'let' block

actually, why did you decide to change the previous version? I liked it

Reply:
Sorry, It is my mistake, I did not understand you clearly